### ISSUE TITLE
Integrating OWScholar with owmeta 

### ISSUE THREAD

**Issue by VahidGh** [openworm/openworm-scholar#8]
Develop a module to add a reference to the owmeta.

**Comment by VahidGh** [comment]
@mwatts15, I also can help with this if there could be any guideline I can follow.

**Comment by VahidGh** [comment]
One possible approach could be to start with the current library available in Mendeley:
https://www.mendeley.com/community/openworm/documents/

I could export a .bib file containing all references (some have keywords).
There are also some api available that one can use to manage it via some Python code:
https://github.com/Mendeley/mendeley-api-python-example

Then we can import these into owmeta.

As for new articles, the Semantic Scholar provided by the Allen Institute has some great features satisfying most of our needs. It also provides some api although not mature enough:
https://api.semanticscholar.org

If we could find new articles using Semantic Scholar, alert in a portal, and save the curated ones in owlet, then we could easily export those needed in GitHub pages like this:
https://github.com/openworm/worm-math-book/blob/main/references.bib

**Comment by VahidGh** [comment]
@slarson, Do you think it would be better to continue with Mendeley or use such an approach and move our reference management system to owmeta plus a curation interface or even a mixed method?

**Comment by mwatts15** [comment]
@VahidGh the openworm-scholar "bot" hasn't been running, and I'm not sure it still works. Before going forward with integrating with owmeta, I want to review the goals for openworm-scholar. In particular, I want to make sure we have useful ways to query and access references -- most of what I did was about subscribing for updates on new publications and posting publications, but I'm not clear anymore on use-cases for what happens after someone finds a publication they want to make reference to.

**Comment by VahidGh** [comment]
@mwatts15 That's right. I'm familiar with the goals of the bot we aimed to work on.
What I'm now suggesting is a more general-purposed one which can work along with the bot for managing and alerting new articles.
Beside that, it can help improving our annotations in owmeta. I mean having such a reference management system, one can easily look for references claimed for any parameter used in our models.
Plus, it can be used as a library for those interested in reading related articles to different OW projects and get updated by new articles.
For example, for such a [use case](https://github.com/openworm/worm-math-book/discussions/3) would be great if we could have an integrated reference management system which is being updated systematically (I mean for both fetching data and inserting new data when one finds new related parameter in an article). Here we need some tag management as well so that one can find related articles from e.g. owmeta easily using keywords (this could be integrated with Semantics Scholar or Mendeley via their APIs).

As for integration with owmeta, for the annotation use case it would be really demanding. However, as a reference management system, we could use some bundle as you suggested previously.

**Comment by VahidGh** [comment]
We can even implement a new/another bot for GitHub. 
In this way, every new related publication could be alerted as an issue or a new entry for discussions.
Then the related curator decides where it could be useful. 
Contributors can also discuss how this new publication can benefit OW.
We can start this approach using current publications we have in Mendeley.

**Comment by VahidGh** [comment]
In order to avoid duplications and keep track of annotations we certainly need something like owmeta.

**Comment by mwatts15** [comment]
@VahidGh There's some work to do to allow for an owmeta Bundle to be a primary target for references curated with openworm-scholar. With a Bundle, you're grouping together a bunch of different data, such as all of the data published for a research study, into a versioned snapshot. So, we'd need a way to generate bundle versions for every new reference and change to an existing curated reference. I've got one notional scenario that would make use of this [here](https://github.com/orgs/openworm/projects/12#card-61180286). It needs to be trimmed down, but the key point is that I don't think curators, for the most part, should need to have a concern for the fact that there's a Bundle of references.

I think a specialized [Bundle Loader](https://github.com/openworm/owmeta-core/blob/develop/owmeta_core/bundle/loaders/__init__.py) (and AccessorConfig) would work for this. Each curated reference would be in its own RDF document, probably an N3 file, in a Git repository, and the Bundle Loader would gather all of these up, on-demand, into the Bundle format. Bundle versions are numeric, so we'd want to ensure a linear graph for the commits. Since the commits would be referenced by number, it would be good to automatically generate numerical Git tags for each change (e.g., r1, r2, ...). (I know some project that did this, maybe Blender, back in the day, but I can't find evidence of it now.) Honestly, support for this format would be good generally to reduce the overhead of creating bundles.
